cpp/src/arrow/dataset/file_parquet.h

```diff
@@ -37,24 +37,31 @@ class ARROW_DS_EXPORT ParquetWriteOptions : public FileWriteOptions {
   std::string file_type() const override;
 };
 
-class ARROW_DS_EXPORT ParquetFragment : public FileBasedDataFragment {
- public:
-  bool splittable() const override { return true; }
-};
-
 /// \brief A FileFormat implementation that reads from Parquet files
 class ARROW_DS_EXPORT ParquetFileFormat : public FileFormat {
  public:
-  std::string name() const override;
+  std::string name() const override { return "parquet"; }
 
   /// \brief Return true if the given file extension
-  bool IsKnownExtension(const std::string& ext) const override;
+  bool IsKnownExtension(const std::string& ext) const override {
+    return ext == "par" || ext == name();
+  }
 
   /// \brief Open a file for scanning
   Status ScanFile(const FileSource& location, std::shared_ptr<ScanOptions> scan_options,
                   std::shared_ptr<ScanContext> scan_context,
                   std::unique_ptr<ScanTaskIterator>* out) const override;
 };
 
+class ARROW_DS_EXPORT ParquetFragment : public FileBasedDataFragment {
+ public:
+  ParquetFragment(const FileSource& location, std::shared_ptr<ScanOptions> options)
+      : FileBasedDataFragment(location, std::make_shared<ParquetFileFormat>(), options) {}
+
+  bool splittable() const override { return true; }
+
+  std::shared_ptr<ScanOptions> scan_options() const override { return NULLPTR; }
```

Comment:
It's curious to have the scan options be a property of the fragment rather than something that's passed in when the scan is initiated (i.e. when the function currently called "GetTasks" -- but maybe better "StartScan" or similar -- is called)

Reply:
I think it's reasonable; a data source might pass modified scan options to some fragments (for example if one fragment is missing a column shared by the others, the information about how to reconcile that would probably be in the scan options.) @fsaintjacques ?

Reply:
I think we'll have to pass some information from the dataset, maybe not as a scan option. But note that this was added because scan_options was part of the DataFragment original interface.